gdnative-core/src/export/property/hint.rs

```diff
@@ -116,12 +116,22 @@ where
 /// ```
 #[derive(Clone, Eq, PartialEq, Debug, Default)]
 pub struct EnumHint {
-    values: Vec<String>,
+    values: Vec<EnumHintEntry>,
 }
 
 impl EnumHint {
     #[inline]
     pub fn new(values: Vec<String>) -> Self {
+        let values = values.into_iter().map(EnumHintEntry::new).collect();
+        EnumHint { values }
+    }
+
+    #[inline]
+    pub fn with_values(values: Vec<(String, i64)>) -> Self {
+        let values = values
+            .into_iter()
+            .map(|(key, value)| EnumHintEntry::with_value(key, value))
+            .collect();
         EnumHint { values }
     }
```

Comment:
Would it be better to simply expose the `EnumHintEntry` interface directly here instead? `fn with_entries(values: Vec<EnumHintEntry>)`